src/rstar.jl

```diff
@@ -84,12 +87,22 @@ function rstar(rng::Random.AbstractRNG, classif::MLJModelInterface.Supervised, c
     return rstar(rng, classif, x, y; kwargs...)
 end
 
-function rstar_score(rng::Random.AbstractRNG, classif::MLJModelInterface.Probabilistic, fitresult, xtest, ytest)
-    pred = get.(rand.(Ref(rng), MLJModelInterface.predict(classif, fitresult, xtest)))
-    return mean(((p,y),) -> p == y, zip(pred, ytest))
+function rstar_distribution(pred::T, ytest::T) where {T<:AbstractVector}
+    mean_accuracy = mean(zip(pred, ytest)) do (p, y)
+        p == y
+    end
+    nclasses = length(MMI.classes(ytest))
+    return Dirac(nclasses * mean_accuracy)
 end
 
-function rstar_score(rng::Random.AbstractRNG, classif::MLJModelInterface.Deterministic, fitresult, xtest, ytest)
-    pred = MLJModelInterface.predict(classif, fitresult, xtest)
-    return mean(((p,y),) -> p == y, zip(pred, ytest))
+function rstar_distribution(
+    pred::AbstractVector{<:UnivariateDistribution},
+    ytest::AbstractVector,
+)
+    # Compute probabilities of Poisson Binomial distribution with support `0:length(ytest)`.
+    probs = Distributions.poissonbinomial_pdf(map(pdf, pred, ytest))
```

Comment:
This algorithm has quadratic complexity in the number of test data points. However, the mean and variance of the distribution can be computed from the predicted probabilities with linear complexity.